### PR TITLE
Add consul stage

### PR DESCRIPTION
**Pull request by funkypenguin**
Hey guys,

This PR adds a new stage, comprising of 2 tasks, plus necessary params. We install consul in a secure (both RPC and Gossip) and HA fashion. It's modelled on the etcd deployment pattern.

This is a pre-req to a coming PR to deploy an HA Vault installation on top of the consul install.

Here's an example of the end result:

```
consul.json  env.sh
[root@mn1 ~]# source /etc/consul.d/env.sh
[root@mn1 ~]# consul operator raft list-peers
Node                                 ID                                    Address             State     Voter  RaftProtocol
mn3.kube-cluster.funkypenguin.co.nz  ba308fdc-ece3-35b3-4895-f8a349c24faf  192.168.33.13:8300  follower  true   3
mn2.kube-cluster.funkypenguin.co.nz  bdfd0897-52a0-2d1d-8db5-ffbe09a102de  192.168.33.12:8300  leader    true   3
mn1.kube-cluster.funkypenguin.co.nz  0d7b5d27-e864-1cf3-9123-d55f136da9e6  192.168.33.11:8300  follower  true   3
[root@mn1 ~]#
```

Cheers!
D

### CODE REVIEW

File: krib/templates/consul-install.sh.tmpl [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  download -L https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_linux_amd64.zip -o ${TMP_DIR}/consul_${CONSUL_VERSION}_linux_amd64.zip
+fi
+
+echo "Install consul version: ${CONSUL_VERSION}"
+yum -y install unzip || echo "Unzip already installed"
+unzip -o ${TMP_DIR}/consul_${CONSUL_VERSION}_linux_amd64.zip -d ${INSTALL_DIR}
+
+
+
+cat >/etc/systemd/system/consul.service <<EOF
```
Comment (galthaus) [inline krib/templates/consul-install.sh.tmpl]: I'm not going to stop this PR, but there is another feature of tasks that you might want to use.
Inside the template list of a task, you can specify a path (the path is left blank to cause the runner to execute the template).  The contents of the template is expanded and stored at the location in path.  This allows you to put things like this and the next file into a template (or inlined) and put in place outside of the script.  This cleans up the script parts.  We aren't consistent about using it and it doesn't work if you use calculated values in the EOF section.  In these cases, it appears you don't.

Comment (galthaus) [reply krib/templates/consul-install.sh.tmpl]: You actually already use in the consul config task.  So.  Up to you.

**Comment by galthaus** [comment]
I would love to pull some of these apart.  They aren't really krib specific.  Kinda like ansible roles.  One day.

File: krib/templates/consul-install.sh.tmpl (hunk 1 above)
Comment (funkypenguin) [reply krib/templates/consul-install.sh.tmpl]: Aah, the original motivation for doing this _was_ calculated values, but I couldn't get it working properly (the old interpolation issue). I'll clean these up slightly as part of the next PR (the addition of a consul-agent, which is _also_ a pre-req to vault!)

**Pull request merged**